examples.md

```diff
@@ -416,6 +417,30 @@ Esy allows you to export built dependencies and import pre-built dependencies.
       if: steps.restore-cache.outputs.cache-hit != 'true'
 ```
 
+## OCaml/Reason - opam
+
+This example requires you to have `opam.locked` file which you can generate with `opam lock .`.
+
+It's worth noting that even if lock file was change, it might be optimal to restore previous cache, so `hashFiles` in the key and if statement in "Save opam cache" step are optional.
+```yaml
```

Comment:
Are the separate steps for `restore` and `save` needed here?

The default action will add a save post step automatically for you and handle checking the `cache-hit` output.

```yaml
    - name: Cache opam
      uses: actions/cache@v4
      with:
        path: _opam
        key: ${{ runner.os }}-opam-${{ hashFiles('opam.locked') }}

    - name: Install dependencies
      run: opam install . --locked --deps-only -y
```